airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx

```diff
@@ -0,0 +1,205 @@
+              </Text>
+              <Controller
+                control={control}
+                name="logicalDate"
+                render={({ field }) => (
+                  <Input
+                    {...field}
+                    placeholder="yyyy-mm-ddThh:mm"
+                    size="sm"
+                    type="datetime-local"
+                  />
+                )}
+              />
+            </Box>
+
+            <Box>
+              <Text fontSize="md" mb={2}>
+                Run ID
+              </Text>
+              <Controller
+                control={control}
+                name="runId"
+                render={({ field }) => (
+                  <Input
+                    {...field}
+                    placeholder="Run id, optional - will be generated if not provided"
+                    size="sm"
+                  />
+                )}
+              />
+            </Box>
+
+            <Box mb={9}>
+              <Text fontSize="md" mb={2}>
+                Configuration JSON
+              </Text>
+              <Controller
+                control={control}
+                name="configJson"
+                render={({ field }) => (
+                  <Box>
+                    <CodeMirror
+                      {...field}
+                      basicSetup
+                      extensions={[json(), autocompletion(), lineNumbers()]}
+                      height="200px"
+                      onChange={(value) => {
+                        field.onChange(value); // Update the form state
+                        try {
+                          // Attempt to parse the value as JSON
+                          JSON.parse(value);
+                          setJsonError(undefined); // Clear error if JSON is valid
+                        } catch {
+                          setJsonError("Invalid JSON format."); // Set error message if invalid
+                        }
+                      }}
+                      style={{
+                        border: "1px solid #CBD5E0",
+                        borderRadius: "8px",
+                        outline: "none",
+                        padding: "2px",
+                      }}
+                      theme={
+                        colorMode === "dark"
+                          ? (githubDark as Extension)
+                          : (githubLight as Extension)
+                      }
+                    />
+                    {jsonError! ? (
+                      <Box color="red.500" mt={2}>
+                        <Text fontSize="sm">{jsonError}</Text>
+                      </Box>
+                    ) : undefined}
+                  </Box>
+                )}
+              />
+            </Box>
+          </VStack>
+        ) : undefined}
+      </VStack>
+
+      <Box as="footer" display="flex" justifyContent="flex-end">
+        <HStack w="full">
+          {hasFormChanged() && <Button onClick={() => reset()}>Reset</Button>}
```

Comment:
```suggestion
          {hasFormChanged() && <Button onClick={reset}>Reset</Button>}
```

Reply:
Showing type mismatch error.
`Type 'UseFormReset<DagParams>' is not assignable to type 'MouseEventHandler<HTMLButtonElement>'.
  Types of parameters 'values' and 'event' are incompatible.
    Type 'MouseEvent<HTMLButtonElement, MouseEvent>' is not assignable to type 'DagParams | { configJson?: { [x: string]: unknown; } | undefined; dagId?: string | undefined; logicalDate?: string | undefined; runId?: string | undefined; } | ResetAction<DagParams> | undefined'.`